testing/tests/test_custom_single_json_retriever.py

```diff
@@ -1,8 +1,72 @@
-import pytest
 import filecmp
+import json
+import perftestnotebook
+import pytest
+import yaml
+from utilities import get_nested_values
+from functools import reduce
+from operator import add
+
+
+class TestTransformer(object):
+    def default_transformer(self, test_config_filepath, target_config_filepath):
+        with open(test_config_filepath, "r") as f, open(target_config_filepath, "r") as g:
+
+            test_config = yaml.safe_load(f)
+            target_config = yaml.safe_load(g)
+            assert "file_groups" in test_config
+            assert "file_groups" in target_config
+
+            test_ptnb = perftestnotebook.PerftestNotebook(test_config["file_groups"], test_config)
+            target_ptnb = perftestnotebook.PerftestNotebook(
+                target_config["file_groups"], target_config
+            )
+
+            target_resource_files = target_ptnb.parse_file_grouping(
+                list(target_ptnb.file_groups.values())
+            )
+
+            target_resource_files = get_nested_values(target_resource_files)
+
+            test_output_filepath = test_ptnb.parse_output()
+            target_output_filepath = target_ptnb.parse_output()
+            assert filecmp.cmp(test_output_filepath, target_output_filepath)
```

Comment:
If the files are different, this is going to fail immediately, and it won't give us much information about why. Move this to the end of the test.